File: crates/deepsec-cli/src/commands/enrich.rs
use crate::config_loader::Context;
use anyhow::Result;
use clap::Args as ClapArgs;
use colored::Colorize;
use deepsec_processor::{EnrichOptions, run_enrich};

#[derive(Debug, ClapArgs)]
pub struct Args {
    #[arg(long)]
    pub project_id: String,
    /// Only enrich paths starting with this prefix.
    #[arg(long)]
    pub filter: Option<String>,
    /// Max recent committers to record per file.
    #[arg(long, default_value_t = 5)]
    pub max_committers: usize,
    /// Re-enrich files that already have gitInfo.
    #[arg(long)]
    pub force: bool,
}

pub fn run(args: Args, ctx: &Context) -> Result<()> {
    let proj = ctx.project(&args.project_id)?;
    let opts = EnrichOptions {
        project_id: args.project_id.clone(),
        project_root: proj.root,
        data_root: ctx.data_root.clone(),
        filter_prefix: args.filter,
        max_committers: args.max_committers,
        force: args.force,
    };
    let out = run_enrich(opts)?;
    println!(
        "{} enriched={} skipped={}",
        "enrich".bold().green(),
        out.files_enriched,
        out.files_skipped
    );
    Ok(())
}

File: internal/cli/commands/enrich.go
package commands

import (
	"fmt"

	"github.com/noeljackson/deepsec/internal/cli"
	"github.com/noeljackson/deepsec/internal/processor"
	"github.com/spf13/cobra"
)

// NewEnrichCmd runs the git-log enrichment pass.
func NewEnrichCmd(loader func() (*cli.Context, error)) *cobra.Command {
	var projectID, filter string
	var maxCommitters int
	var force bool
	cmd := &cobra.Command{
		Use:   "enrich",
		Short: "Populate FileRecord.gitInfo.recentCommitters via git log",
		RunE: func(cmd *cobra.Command, _ []string) error {
			ctx, err := loader()
			if err != nil {
				return err
			}
			proj, err := ctx.Project(projectID)
			if err != nil {
				return err
			}
			out, err := processor.Enrich(processor.EnrichOptions{
				ProjectID:     projectID,
				ProjectRoot:   proj.Root,
				DataRoot:      ctx.DataRoot,
				FilterPrefix:  filter,
				MaxCommitters: maxCommitters,
				Force:         force,
			})
			if err != nil {
				return err
			}
			fmt.Printf("enrich enriched=%d skipped=%d\n", out.FilesEnriched, out.FilesSkipped)
			return nil
		},
	}
	cmd.Flags().StringVar(&projectID, "project-id", "", "Project id (required)")
	_ = cmd.MarkFlagRequired("project-id")
	cmd.Flags().StringVar(&filter, "filter", "", "Path-prefix filter")
	cmd.Flags().IntVar(&maxCommitters, "max-committers", 5, "Max recent committers per file")
	cmd.Flags().BoolVar(&force, "force", false, "Re-enrich files that already have gitInfo")
	return cmd
}
